lib/mayaUsd/fileio/shading/shadingModeUseRegistry.cpp

```diff
@@ -271,17 +271,19 @@ class UseRegistryShadingModeExporter : public UsdMayaShadingModeExporter
 
                 // We pass in the type of the plug on the other side to allow the export code to
                 // add conversion nodes as required.
-
-                const TfToken srcPlugName
-                    = TfToken(UsdMayaShadingUtil::GetStandardAttrName(srcPlug, false));
-                UsdAttribute srcAttribute = srcShaderInfo->GetShadingAttributeForMayaAttrName(
-                    srcPlugName, MayaUsd::Converter::getUsdTypeName(dstPlug));
-
                 const TfToken dstPlugName
                     = TfToken(UsdMayaShadingUtil::GetStandardAttrName(dstPlug, false));
                 UsdAttribute dstAttribute = dstShaderInfo->GetShadingAttributeForMayaAttrName(
                     dstPlugName, MayaUsd::Converter::getUsdTypeName(srcPlug));
 
+                UsdAttribute srcAttribute;
+                if (dstAttribute) {
```

Comment:
Why was this change necessary?

Reply:
Because the connection on the Maya side is to a float3 opacity, while the connection USD side is to a float opacity. We need this to get the proper destination type.